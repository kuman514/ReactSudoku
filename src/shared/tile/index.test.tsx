import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import { beforeEach, describe, expect, it, vi } from 'vitest';

import { Tile } from '.';

describe('Tile', () => {
  beforeEach(() => {
    cleanup();
  });

  it('should have a snapshot match for a colored tile', () => {
    const { container } = render(
      <Tile isColored>This is a colored tile.</Tile>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a snapshot match for a clear tile', () => {
    const { container } = render(<Tile>This is an clear tile.</Tile>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be clickable', async () => {
    const mockFn = vi.fn();
    render(
      <Tile ariaLabel="test-tile" onClick={mockFn}>
        Click Test
      </Tile>
    );
    fireEvent.click(await screen.findByLabelText('test-tile'));
    expect(mockFn).toBeCalledTimes(1);
  });

  it('should be not clickable when it is disabled', async () => {
    const mockFn = vi.fn();
    render(
      <Tile ariaLabel="test-tile" onClick={mockFn} isDisabled>
        Click Test
      </Tile>
    );
    fireEvent.click(await screen.findByLabelText('test-tile'));
    expect(mockFn).toBeCalledTimes(0);
  });
});
