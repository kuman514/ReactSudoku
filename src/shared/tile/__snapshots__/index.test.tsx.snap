// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tile > should have a snapshot match for a clear tile 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1n3uo6c-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  This is an clear tile.
</button>
`;

exports[`Tile > should have a snapshot match for a colored tile 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-4x02tf-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  This is a colored tile.
</button>
`;
